source/Octopus.Client/OctopusAsyncRepository.cs

```diff
@@ -134,5 +156,138 @@ public OctopusAsyncRepository(IOctopusAsyncClient client)
         public IWorkerRepository Workers { get; }
         public IScopedUserRoleRepository ScopedUserRoles { get; }
         public IUserPermissionsRepository UserPermissions { get; }
+
+        public async Task<IOctopusAsyncRepository> ForSpaceContext(string spaceId)
+        {
+            ValidateSpaceId(spaceId);
+            var spaceRoot = await LoadSpaceRootResource(Client, RootDocument, spaceId);
+            return new OctopusAsyncRepository(Client, RootDocument, spaceRoot, SpaceContext.SpecificSpace(spaceId));
```

Comment:
`RootDocument` is shared

Reply:
It would be better to pass the root repository around instead of the root document. That way, if you refresh the root repository, this change will propagate to all linked child repositories.